controllers/src/main/java/io/syndesis/controllers/integration/IntegrationController.java

```diff
@@ -51,22 +56,33 @@
 
     private final DataManager dataManager;
     private final EventBus eventBus;
-    private final ConcurrentHashMap<Integration.Status, StatusChangeHandlerProvider.StatusChangeHandler> handlers = new ConcurrentHashMap<>();
+    private final Map<IntegrationState, StatusChangeHandler> draftHandlers;
+    private final Map<IntegrationState, StatusChangeHandler> deployedHandlers;
+
     private final Set<String> scheduledChecks = new HashSet<>();
     private ExecutorService executor;
     private ScheduledExecutorService scheduler;
 
     private static final long SCHEDULE_INTERVAL_IN_SECONDS = 60;
 
     @Autowired
-    public IntegrationController(DataManager dataManager, EventBus eventBus, StatusChangeHandlerProvider handlerFactory) {
+    public IntegrationController(DataManager dataManager, EventBus eventBus, StatusChangeHandlerProvider handlerProvider) {
         this.dataManager = dataManager;
         this.eventBus = eventBus;
-        for (StatusChangeHandlerProvider.StatusChangeHandler handler : handlerFactory.getStatusChangeHandlers()) {
-            for (Integration.Status status : handler.getTriggerStatuses()) {
-                this.handlers.put(status, handler);
-            }
-        }
+        this.draftHandlers = new HashMap<>();
+        this.deployedHandlers = new HashMap<>();
+
+        handlerProvider.getDraftStatusChangeHandlers().forEach(h ->
+           draftHandlers.putAll(h.getTriggerStatuses()
+               .stream()
+               .collect(Collectors.toMap(t -> t, t -> h)))
+        );
+
+        handlerProvider.getDeployedStatusChangeHandlers().forEach(h ->
```

Comment:
status2handler mapping func could be moved in a single method as it duplicated here.